### PR TITLE
Fix: datetime default values

### PR DESCRIPTION
**Pull request by Meldiron**
Library was not forcing datetime filter on datetime-type attributes, even tho it is necessary for proper conversion on adapter level. This PR forces the adapter internally, and introduces tests to ensure the use case works.

### CODE REVIEW

File: src/Database/Database.php [hunk 1]
```diff
@@ -785,7 +789,7 @@ public function updateAttributeDefault(string $collection, string $id, $default
                 case self::VAR_INTEGER:
                 case self::VAR_FLOAT:
                 case self::VAR_BOOLEAN:
-                    if ($attribute->getAttribute('type') !== \gettype($default)) {
+                        if ($attribute->getAttribute('type') !== \gettype($default) && $attribute->getAttribute('type') !== self::VAR_DATETIME) {
```
Comment (TorstenDittmann) [inline src/Database/Database.php:792]:
```suggestion
                    if ($attribute->getAttribute('type') !== \gettype($default) && $attribute->getAttribute('type') !== self::VAR_DATETIME) {
```

File: src/Database/Database.php [hunk 2]
```diff
@@ -635,7 +639,7 @@ public function createAttribute(string $collection, string $id, string $type, in
                 case self::VAR_INTEGER:
                 case self::VAR_FLOAT:
                 case self::VAR_BOOLEAN:
-                    if ($type !== \gettype($default)) {
+                    if ($type !== \gettype($default) && $type !== self::VAR_DATETIME) {
```
Comment (TorstenDittmann) [inline src/Database/Database.php:642]: I feel like the condition for datetime should only apply for the case of a string right?

Otherwise its hard to follow why the condition is there.

**Comment by Meldiron** [comment]
Problem addressed in a better way in this PR: 
https://github.com/utopia-php/database/pull/185
